File: ffacg-vue/src/stores/anime.ts
import axios from "axios"
import { defineStore } from "pinia"

export const useAnimeStore = defineStore("anime", () => {

    const get_animes = async (page: number) => {
        const res = await axios.get("https://api.animes.garden/resources", {
            params: { type: '动画', page: page, pageSize: 20 }
        })
        if (res.data.status === 'OK') {
            let arr = res.data.resources as Anime[]
            let uuid = 0;
            const r = arr.map((v) => ({ ...v, id: (uuid++).toString() }))
            return r as Anime[]
        }
    }

    return { get_animes }
})


export interface Anime {
    id: string
    title: string
    size: number
    magnet: string
    tracker: string
    fansub: Fansub
    publisher: Publisher
}

export interface Fansub {
    id: string
    name: string
    avatar: string
}

export interface Publisher {
    id: string
    name: string
    avatar: string
}

export interface AnimeHistory {
    id: string
    anime: Anime
    time: Date
}

export interface AnimeMask {
    id: string
    anime: Anime
    group: string
}

File: src/model/anime.rs
use serde::{Deserialize, Serialize};
use sqlx::prelude::FromRow;


#[derive(Serialize, Deserialize, Debug, FromRow)]
pub struct Anime {
    pub id: Option<String>,
    pub title: Option<String>,
    pub size: Option<i32>,
    pub magnet: Option<String>,
    pub tracker: Option<String>,
    pub fansub: Option<Fansub>,
    pub publisher: Option<Publisher>,
}

#[derive(Serialize, Deserialize, Debug, FromRow)]
pub struct Fansub {
    pub id: Option<String>,
    pub name: Option<String>,
    pub avatar: Option<String>,
}

pub type Publisher = Fansub;

#[derive(Serialize, Deserialize, Debug, FromRow)]
pub struct AnimeHistory {
    pub id: Option<String>,
    pub anime: Option<Anime>,
    pub time: Option<i64>,
}

#[derive(Serialize, Deserialize, Debug, FromRow)]
pub struct AnimeMask {
    pub id: Option<String>,
    pub anime: Option<Anime>,
    pub group_name: Option<String>,
}